Cesium.CodeGen/Ir/Expressions/Values/LValueArrayElement.cs

```diff
@@ -59,13 +59,34 @@ private void EmitPointerMoveToElement(IEmitScope scope)
         method.Emit(OpCodes.Add);
     }
 
-    private static IType GetBaseType(InPlaceArrayType valueType)
+    private static IType GetBaseType(IType valueType)
+    {
+        var baseType = valueType switch
+        {
+            InPlaceArrayType arrayType => arrayType.Base,
+            PointerType pointerType => pointerType.Base,
+            _ => throw new AssertException("Array or pointer type expected.")
+        };
+        if (baseType is InPlaceArrayType or PointerType)
+        {
+            return GetBaseType(baseType);
+        }
+
+        return baseType;
+    }
```

Comment:
I feel like this all is broken.

`LValueArrayElement.GetValueType()` should return the type of the array element, while it currently returns the array type itself.

While this `GetBaseType()` should only perform one step of indirection and not be recursive!

In your PR and even before that, it seems to always remove all the indirection levels and I don't understand this behavior at all.

Reply:
You know, you are right. Thanks your for keeping me in check.